libs/payments/paypal/src/lib/paypal.service.spec.ts

```diff
@@ -0,0 +1,63 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { Test, TestingModule } from '@nestjs/testing';
+import { faker } from '@faker-js/faker';
+
+import { CurrencyManager } from '@fxa/payments/currency';
+
+import { PayPalManager } from './paypal.manager';
+import { PayPalService } from './paypal.service';
+import { ResultPaypalCustomerFactory } from './paypalCustomer/paypalCustomer.factories';
+import { PaypalCustomerManager } from './paypalCustomer/paypalCustomer.manager';
+
+describe('PayPalService', () => {
+  let mockPayPalService: PayPalService;
+
+  beforeAll(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        {
+          provide: PayPalManager,
+          useValue: {
+            createBillingAgreement: jest.fn(),
+            getBillingAgreement: jest.fn(),
+          },
+        },
+        {
+          provide: CurrencyManager,
+          useValue: {
+            assertCurrencyCompatibleWithCountry: jest.fn(),
+          },
+        },
+        {
+          provide: PaypalCustomerManager,
+          useValue: {
+            createPaypalCustomer: jest.fn(),
+          },
+        },
+        PayPalService,
+      ],
+    }).compile();
+
+    mockPayPalService = module.get<PayPalService>(PayPalService);
+  });
+
+  describe('createBillingAgreement', () => {
+    it('creates a billing agreement', async () => {
+      const mockNewPaypalCustomer = ResultPaypalCustomerFactory();
+
+      const uid = faker.string.sample();
+      const token = faker.string.sample();
+      const currency = faker.string.sample();
+      const options = { uid, token, currency };
+
+      mockPayPalService.createBillingAgreement = jest
+        .fn()
+        .mockResolvedValueOnce(mockNewPaypalCustomer);
+
+      const result = await mockPayPalService.createBillingAgreement(options);
+      expect(result).toEqual(mockNewPaypalCustomer);
```

Comment:
By mocking createBillingAgreement on line 55, you've made this test a non-op. Your lines 59 and 60 don't do anything, and the code under test (in the library) is never called or tested.